python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -251,184 +256,133 @@ def __init__(self, installation_root, logger, core_version=None):
                                   e.g. you can run the updater as a totally separate thing.
         :param logger: Logger to send output to.
         :param core_version: Version of the core to update to. If None, the core will be updated to the latest
-                             version.
+                             version. Defaults to None.
         """
         self._log = logger
 
-        (sg_app_store, script_user) = shotgun.create_sg_app_store_connection()
-        self._sg = sg_app_store
-        self._sg_script_user = script_user
+        from tank_vendor.shotgun_deploy import Descriptor, create_descriptor
 
-        self._local_sg = shotgun.get_sg_connection()
-        self._update_version = self.__get_core_version(core_version)
+        self._current_core_descriptor = create_descriptor(
```

Comment:
Damn, sorry i realise now that this tank command is `Action.GLOBAL` so hence there is no pc available. Sorry!